ibis/pandas/aggcontext.py

```diff
@@ -252,19 +252,41 @@ def agg(self, grouped_data, function, *args, **kwargs):
         pass
 
 
-def make_applied_function(function, args=None, kwargs=None):
+def wrap_for_apply(function, args, kwargs):
+    """Wrap a function so that it may be used with Pandas `apply`"""
     assert callable(function), 'function {} is not callable'.format(function)
 
     @functools.wraps(function)
-    def apply(
-        data,
-        function=function,
-        args=args if args is not None else (),
-        kwargs=kwargs if kwargs is not None else {},
-    ):
+    def wrapped_func(data, function=function, args=args, kwargs=kwargs):
+        return function(data, *args, **kwargs)
+
+    return wrapped_func
+
+
+def wrap_for_agg(function, args, kwargs):
+    """Wrap a function so that it may be used with Pandas `agg`.
+
+    In some cases, Pandas `agg` will try to behave like Pandas `apply`.
+    `function` will be wrapped in such a way that Pandas will be forced
+    not to do this (in other words, forced to treat `function` as an
```

Comment:
How did you enforce that `function` is treated as an aggregation function? It looks like you are enforcing the type of input data to be Series. It's not clear to me if that is what you mean by "forced to treat `function` as an
aggregation functionn"

Reply:
Yes the type-checking / error-raising logic is what's forcing `function` to be treated as an aggregation function. I re-wrote the docstring for `wrap_for_agg` to help explain this, please check:
https://github.com/ibis-project/ibis/pull/2410/files#diff-7aa946a96f18fd1e3be4bfa4f37efd86R266-R284